crates/wash-lib/src/start/github/api.rs

```diff
@@ -0,0 +1,290 @@
+use anyhow::Error;
+use chrono::{DateTime, Utc};
+use semver::Version;
+use serde::{Deserialize, Serialize};
+use wasmcloud_core::tls::NativeRootsExt;
+
+type DateTimeUtc = DateTime<Utc>;
+
+/// GitHub page max https://docs.github.com/en/rest/releases/releases?apiVersion=2022-11-28#list-releases
+const GITHUB_PER_PAGE: u32 = 100;
+
+const VERSION_FETCHER_CLIENT_USER_AGENT: &str =
+    concat!(env!("CARGO_PKG_NAME"), "/", env!("CARGO_PKG_VERSION"));
+
+/// get_chronologically_sorted_releases_of_after returns with a list of chronologically ordered (latest first)
+/// with the provided after_version being excluded
+async fn get_chronologically_sorted_releases_of_after(
+    owner: &str,
+    repo: &str,
+    after_version: &semver::Version,
+) -> Result<Vec<GitHubRelease>, anyhow::Error> {
```

Comment:
I find the naming of these functions just a little confusing, my suggestion would be to drop a bit of wording from the function in favor of doc comments.
```suggestion
/// get_chronologically_sorted_releases returns with a list of chronologically ordered (latest first)
/// releases that are more recent than the provided `after_version`.
async fn get_chronologically_sorted_releases(
    owner: &str,
    repo: &str,
    after_version: &semver::Version,
) -> Result<Vec<GitHubRelease>, anyhow::Error> {
```

Totally open to suggestions here, this was more of a nit and since it's not a public function I'm not as concerned about it